### PR TITLE
Android/Contextual: Fix auto-attach behavior

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/1/137249556945/project/488551667048375/task/1216113532784491?focus=true 
Tech Design URL (if applicable): 

### Description
Refines page-context attachment behavior in the Duck.ai contextual sheet, keying off the user's automatic-context-attachment setting.
  - **Auto-attach under `contextualSheetImprovements`:** when the setting is on, page context is attached on receipt and the quick action goes straight to `SUBMIT_SUMMARIZE`, skipping the `ASK_ABOUT_PAGE` step.
  When the setting is off, the manual `ASK_ABOUT_PAGE` flow is unchanged.
  - **Reopen on a different URL (auto-attach off):** if the sheet is closed and reopened on a URL different from the attached one, the stale attachment is dropped instead of silently swapping to the new page; the
  pill reverts to `ASK_ABOUT_PAGE` so the user can re-attach.
  - **Re-attach on navigation (auto-attach on):** a manual removal is now scoped to the URL it was made on — navigating to a different URL re-attaches the new page, while re-receiving the same URL still respects
  the removal.
  - Passive navigation with auto-attach off still leaves the attached context untouched.

### Steps to test this PR
Setup:
 - Run all the tests with contextualNativeInput = FALSE 
 - Run all tests again with contextualNativeInput = TRUE (Manually enable via FF inventory, app restart required)
 
  ### 1. Auto-attach OFF — empty sheet, navigate Page A → Page B (sheet left open)
  - [x] **Automatically Send Page Content** = OFF
  - [x] Open the contextual sheet on Page A 
  - [x] Verify that NO context is attached automatically
  - [x] Without closing the sheet (just unfocus the input), Navigate from Page A → Page B
  - [x] Verify that No context is attached automatically
  - [x] The “Ask About Page” is shown in both cases

  ### 2a. Auto-attach OFF — manually attach Page A, navigate to Page B (sheet left open)
  - [x] **Automatically Send Page Content** = OFF
  - [x] Manually attach Page A (via "Ask about page") — attachment shows Page A
  - [x] Without closing the sheet (just unfocus the input), Navigate from Page A → Page B
  - [x] Page A **stays** attached (chip still shows Page A, not Page B)
  - [x] Remove attachment and select Ask About Page again
  - [x] Verify that the correct url is attached (Page B)

  ### 2b. Auto-attach OFF — manually attach Page A, close, navigate to Page B, reopen on Page B
  - [x] **Automatically Send Page Content** = OFF
  - [x] Manually attach Page A — attachment shows Page A
  - [x] Close the sheet, navigate the browser to Page B, then reopen the sheet on Page B
  - [x] Page A attachment is **cleared** (no attachment shown)
  - [x] The quick-action pill returns to **Ask about page**
  - [x] Tapping **Ask about page** now attaches Page B (current page)

  ### 3. Auto-attach OFF — manually attach Page A, remove, navigate Page A → Page B
  - [x] **Automatically Send Page Content** = OFF
  - [x] Manually attach Page A, then remove it (attachment dismissed) and Close the sheet,
  - [x] Navigate the underlying browser Page A → Page B
  - [x] No context attaches automatically
  - [x] The “Ask About Page” is shown

  ### 4. Auto-attach ON — Page A attached, navigate Page A → Page B
  - [x] **Automatically Send Page Content** = ON
  - [x] Open the sheet on Page A — Page A auto-attaches (attachment shows Page A)
  - [x] Without closing the sheet, Navigate the underlying browser Page A → Page B
  - [x] Chip updates to Page B (Page B attaches)

  ### 5. Auto-attach ON — Page A attached, remove, navigate Page A → Page B
  - [x] **Automatically Send Page Content** = ON
  - [x] Open the sheet on Page A — Page A auto-attaches
  - [x] Remove the attached context (attachment dismissed)
  - [x] Without closing the sheet, Navigate the underlying browser Page A → Page B
  - [x] Page B auto-attaches (chip shows Page B)

  ### 6. Auto-attach ON — submit with Page A context, close/reopen sheet on same Page A
  - [x] **Automatically Send Page Content** = ON
  - [x] On Page A, submit a prompt with Page A context attached — sheet moves to the chat (WEBVIEW)
  - [x] Close the sheet, then reopen it on the same Page A
  - [x] The existing chat/session is restored (no new chat created)
  - [x] Page A context is **not** re-attached as a new chip or re-submitted / duplicated in the conversation

---

> [!NOTE]
> **Medium Risk**
> Changes prompt submission and page-context attachment logic in the contextual AI sheet across navigation and settings; wrong edge-case handling could send stale or missing page content to Duck.ai, but behavior is heavily covered by new unit tests.
> 
> **Overview**
> **Fixes Duck.ai contextual sheet page-context attachment** so behavior matches the **Automatically Send Page Content** setting and navigation/reopen cases.
> 
> The ViewModel now splits **latest browser page** (`currentPageContext`) from a **frozen attachment snapshot** (`attachedPage`) used when submitting prompts, so passive navigation with auto-attach off no longer swaps what an existing chip sends. **`tabId`** is set synchronously in `onSheetOpened` and used from `viewState` (replacing `sheetTabId`); the fragment passes `currentPageContext` and `tabId` into JS callbacks.
> 
> **`onPageContextReceived`** always tracks valid latest context; unsolicited updates in INPUT with auto-attach off skip UI changes but still update `currentPage` for manual attach. With auto-attach on, context can auto-attach and move quick actions to **`SUBMIT_SUMMARIZE`**; with auto-attach off, a URL change while attached drops a stale chip and returns to **`ASK_ABOUT_PAGE`**. User removal is URL-scoped (new URL can re-attach; same URL respects removal). **`onMainBrowserPageFinished`** always requests fresh context in INPUT (not only when auto-attach is on). New chat clears `attachedPage`.
> 
> Tests are expanded/updated for these navigation, reopen, and auto-attach matrix cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1b8d361cd9b5056e58294354d54ebceec1644a3e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by karlenDimla** [comment]
* **#9245** <a href="https://app.graphite.com/github/pr/duckduckgo/Android/9245?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/duckduckgo/Android/9245?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `develop`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/contextual/DuckChatContextualFragment.kt [hunk 1]
```diff
@@ -393,8 +393,8 @@ class DuckChatContextualFragment :
                                             id,
                                             data,
                                             Mode.CONTEXTUAL,
-                                            viewModel.updatedPageContext,
-                                            viewModel.sheetTabId,
+                                            viewModel.currentPageContext,
+                                            viewModel.viewState.value.tabId,
```
[the remark this answers is not in the document]
Comment (karlenDimla) [reply duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/contextual/DuckChatContextualFragment.kt:397]: This is not a real issue.

The reason-gating in `getAIChatPageContext` already prevents the mismatch you'd worry about — the only time currentPage and attachedPage diverge is auto-attach OFF, and in that case `REASON_INIT` returns empty rather than the param. The only path that returns the passed context during a divergence is `REASON_USER_ACTION`, where the user is explicitly attaching the page they're viewing, so the live currentPage is the correct value.

**Review by joshliebe — APPROVED** [review]
Tested all the scenarios, works as expected 🚀

File: duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/contextual/DuckChatContextualViewModel.kt [hunk 2]
```diff
@@ -710,10 +728,14 @@ class DuckChatContextualViewModel @Inject constructor(
...
             }
+            val hasValidContext = isContextValid(currentPageContext)
+            if (hasValidContext) {
+                pageContextState = pageContextState.copy(attachedPage = pageContextState.currentPage)
+            }
             _viewState.update { current ->
                 current.copy(
                     prompt = newPrompt,
-                    showContext = isContextValid(updatedPageContext),
+                    showContext = hasValidContext,
```
[the remark this answers is not in the document]
Comment (karlenDimla) [reply duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/contextual/DuckChatContextualViewModel.kt:738]: Will remove the legacy soon (not used in prod anymore) so this is OK.
